yarn-project/sequencer-client/src/sequencer/sequencer.ts

```diff
@@ -136,7 +137,11 @@ export class Sequencer {
 
       // Process txs and drop the ones that fail processing
       // We create a fresh processor each time to reset any cached state (eg storage writes)
-      const processor = await this.publicProcessorFactory.create(prevGlobalVariables, newGlobalVariables);
+      const processor = await this.publicProcessorFactory.create(
+        prevGlobalVariables,
+        newGlobalVariables,
+        validTxs.flatMap(tx => tx.newContracts),
```

Comment:
Heads up that this allows a tx to access a contract before it's deployed, if the contract is deployed by a tx that occurs later on this same block.

Reply:
That's a good catch. Will need to address @alexghr